### PR TITLE
T2D-268 Fix a segfault caused by out of bounds access.

### PR DESCRIPTION
**Pull request by Zhicheng-Liu**

### CODE REVIEW

File: test/vcf/assembly_checker_integration_test.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ namespace ebi
     {
         std::vector<std::unique_ptr<ebi::vcf::AssemblyCheckReportWriter>> outputs;
         outputs.emplace_back(new ebi::vcf::SummaryAssemblyCheckReportWriter());
-        std::string assembly_report = ebi::vcf::NO_MAPPING;
+            std::string assembly_report = ebi::vcf::NO_MAPPING;
```
Comment (jmmut) [inline test/vcf/assembly_checker_integration_test.cpp]:
```suggestion
        std::string assembly_report = ebi::vcf::NO_MAPPING;
```

File: test/input_files/v4.3/assembly_checker/failed/failed_ena_api_with_mapping/failed_ena_api_with_mapping.vcf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+##fileformat=VCFv4.1
+##FILTER=<ID=PASS,Description="All filters passed">
+##fileDate=20150218
+##reference=ftp://ftp.1000genomes.ebi.ac.uk//vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz
+##source=1000GenomesPhase3Pipeline
+##contig=<ID=AB325691,assembly=b37,length=249250621>
+##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
```
Comment (cyenyxe) [inline test/input_files/v4.3/assembly_checker/failed/failed_ena_api_with_mapping/failed_ena_api_with_mapping.vcf]: Could you add a header entry explaining the reason for failure? If you look at the oldest failed test files you can see an example of how we did it.

File: test/vcf/assembly_checker_integration_test.cpp [hunk 3]
```diff
@@ -182,6 +182,17 @@ namespace ebi
             CHECK_FALSE(ebi::vcf::assembly_checker::check_vcf_ref(vcf_input, vcf_path, fasta, assembly_report, outputs));
         }
 
+        SECTION("with correct mappoing, no fasta provided, using ENA API to retrieve fasta for contigs, but got no sequence, fail")
```
Comment (cyenyxe) [inline test/vcf/assembly_checker_integration_test.cpp]: I'm not sure I understand what this means: "with correct mapping but got no sequence".

File: test/input_files/v4.3/assembly_checker/failed/failed_ena_api_with_mapping/failed_ena_api_with_mapping.vcf (hunk 2 above)
Comment (Zhicheng-Liu) [reply test/input_files/v4.3/assembly_checker/failed/failed_ena_api_with_mapping/failed_ena_api_with_mapping.vcf]: I have added `##CauseOfFailure` tag to indicate what is the VCF designed for in 0edf59b .

File: test/vcf/assembly_checker_integration_test.cpp (hunk 3 above)
Comment (Zhicheng-Liu) [reply test/vcf/assembly_checker_integration_test.cpp]: I have reworded the description in 0edf59b. Hope it is now clearer.

**Pull request merged**